protocols/mp-ecdsa/src/client.rs

```diff
@@ -0,0 +1,241 @@
+        let (execute, force_execute) = self
+            .exec_client_function(move |client| {
+                client.runtime_api().should_execute_new_keygen(at).unwrap_or_default()
+            })
+            .await;
+
+        AnticipatedKeygenExecutionStatus { execute, force_execute }
+    }
+
+    /// Get the next DKG public key
+    pub async fn get_next_dkg_pub_key(
+        &self,
+        header: &B::Header,
+    ) -> Option<(AuthoritySetId, Vec<u8>)> {
+        let at = header.hash();
+        self.exec_client_function(move |client| {
+            client.runtime_api().next_dkg_pub_key(at).unwrap_or_default()
+        })
+            .await
+    }
+
+    pub async fn dkg_pub_key_is_unset(&self, header: &B::Header) -> bool {
+        self.get_dkg_pub_key(header).await.1.is_empty()
+    }
+
+    /// Get the active DKG public key
+    pub async fn get_dkg_pub_key(&self, header: &B::Header) -> (AuthoritySetId, Vec<u8>) {
+        let at = header.hash();
+        self.exec_client_function(move |client| {
+            client.runtime_api().dkg_pub_key(at).unwrap_or_default()
+        })
+            .await
+    }
+
+    /// Get the party index of our worker
+    ///
+    /// Returns `None` if we are not in the best authority set
+    pub async fn get_party_index(&self, header: &B::Header) -> Option<u16> {
+        let public = self.get_authority_public_key();
+        let best_authorities = self.get_best_authorities(header).await;
+        for elt in best_authorities {
+            if elt.1 == public {
+                return Some(elt.0)
+            }
+        }
+
+        None
+    }
+
+    /// Get the best authorities for keygen
+    pub async fn get_best_authorities(&self, header: &B::Header) -> Vec<(u16, AuthorityId)> {
+        let at = header.hash();
+        self.exec_client_function(move |client| {
+            client.runtime_api().get_best_authorities(at).unwrap_or_default()
+        })
+            .await
+    }
+
+    /// Get the next party index of our worker for possible queued keygen
+    ///
+    /// Returns `None` if we are not in the next best authority set
+    pub async fn get_next_party_index(&self, header: &B::Header) -> Option<u16> {
+        let public = self.get_authority_public_key();
+        let next_best_authorities = self.get_next_best_authorities(header).await;
+        for elt in next_best_authorities {
+            if elt.1 == public {
+                return Some(elt.0)
+            }
+        }
+
+        None
+    }
+
+    /// Get the next best authorities for keygen
+    pub async fn get_next_best_authorities(&self, header: &B::Header) -> Vec<(u16, AuthorityId)> {
+        let at = header.hash();
+        self.exec_client_function(move |client| {
+            client.runtime_api().get_next_best_authorities(at).unwrap_or_default()
+        })
+            .await
+    }
+
+    /// Get the signature threshold at a specific block
+    pub async fn get_signature_threshold(&self, header: &B::Header) -> u16 {
```

Comment:
There is no main signature threshold and the client API won't implement this as you have it here.